src/par_iter/internal.rs

```diff
@@ -90,28 +166,35 @@ pub fn bridge<PAR_ITER,C>(mut par_iter: PAR_ITER,
         fn callback<P>(mut self, mut producer: P) -> C::Result
             where P: Producer<Item=ITEM>
         {
-            let producer_cost = producer.cost(self.len);
-            let cost = self.consumer.cost(producer_cost);
-            bridge_producer_consumer(self.len, cost, producer, self.consumer)
+            if producer.weighted() || self.consumer.weighted() {
```

Comment:
Just from a pure API perspective, I am wondering about the sharp transition between "weight" and "not weighted". It is perhaps the right thing to do, but it seems to (very mildly) erode the benefits of "weight vs threshold", since if you add a weight to one subpart, you affect all the parts. But usually people build up their parallel iterators as a unit so that doesn't feel like a big deal.